netebpfext/user/fwp_um.cpp

```diff
@@ -641,6 +640,10 @@ _IRQL_requires_max_(PASSIVE_LEVEL) NTSTATUS FwpsInjectionHandleDestroy0(_In_ HAN
 _IRQL_requires_max_(DISPATCH_LEVEL) NTSTATUS
     FwpsFlowRemoveContext0(_In_ uint64_t flow_id, _In_ UINT16 layer_id, _In_ uint32_t callout_id)
 {
+    if (ebpf_fault_injection_inject_fault()) {
+        return STATUS_NO_MEMORY;
```

Comment:
My understanding for applying fault injection is to ensure callers of the APIs handle the error case so that the application doesn't lead to crash or leak.

In this particular case, I can return STATUS_PENDING.

Reply:
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/fwpsk/nf-fwpsk-fwpsflowremovecontext0 doesn't list STATUS_PENDING, should it? (Why does removing a context need to be asynchronous?  How would you tell when it's complete?)

Reply:
Pasting the 
**Remarks**
If the FwpsFlowRemoveContext0 function returns STATUS_SUCCESS, FwpsFlowRemoveContext0 calls the [flowDeleteFn](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/fwpsk/nc-fwpsk-fwps_callout_flow_delete_notify_fn0) callout function synchronously. **If FwpsFlowRemoveContext0 returns STATUS_PENDING, FwpsFlowRemoveContext0 calls flowDeleteFn asynchronously because an active callout classification is in progress.**

Reply:
However, the ASSERT will ensure the return needs to be SUCCESS. Hence will remove the fault injection from this API.


// https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/fwpsk/nf-fwpsk-fwpsflowremovecontext0
        // Calling FwpsFlowRemoveContext may cause the flowDeleteFn callback on the callout to be invoked synchronously.
        // The net_ebpf_extension_sock_ops_flow_delete function frees the flow context memory and
        // releases reference on the filter_context.
#pragma warning(push)
#pragma warning(disable : 4189) // 'status': local variable is initialized but not referenced
        NTSTATUS status =
            FwpsFlowRemoveContext(flow_parameters->flow_id, flow_parameters->layer_id, flow_parameters->callout_id);
        **ASSERT(status == STATUS_SUCCESS);**
#pragma warning(pop)

Reply:
Thanks for pointing out the Remarks, I will file an issue against the Return Values portion of MSDN.